pkg/db/db.go

```diff
@@ -174,24 +174,33 @@ func (db *DB) SelectIndexByArtifactIDAndGroupID(artifactID, groupID string) (typ
 	return index, nil
 }
 
-func (db *DB) SelectIndexesByArtifactIDAndFileType(artifactID string, fileType types.ArchiveType) ([]types.Index,
+func (db *DB) SelectGroupIDByArtifactIDVersionAndFileType(artifactID, version string, fileType types.ArchiveType) (string,
 	error) {
-	var indexes []types.Index
+	var groupID string
+	var count int
 	rows, err := db.client.Query(`
-		SELECT a.group_id, a.artifact_id, i.version, i.sha1, i.archive_type
-		FROM indices i 
-		JOIN artifacts a ON a.id = i.artifact_id
-        WHERE a.artifact_id = ? AND i.archive_type = ?`,
-		artifactID, fileType)
+		SELECT relevant.group_id, COUNT(relevant.group_id) as count
+		FROM indices i
+		JOIN (SELECT a.id, a.group_id
+        	  FROM indices i
+        	  JOIN artifacts a on a.id = i.artifact_id
+        	  WHERE a.artifact_id = ? AND i.version = ? AND i.archive_type = ?) relevant ON relevant.id = i.artifact_id
+		GROUP BY "group_id"`,
```

Comment:
Do we need `GROUP BY` here? I want to keep the logic in trivy-java-db as minimal as possible. That's why this function returns `indexes`.
We can match the version in SQL as it is simple enough.

Reply:
By the way, what if the latest version is used in the project? The database probably doesn't have it yet, and Trivy will miss it.

Reply:
> Do we need GROUP BY here? I want to keep the logic in trivy-java-db as minimal as possible. That's why this function returns indexes.

I updated query - https://github.com/aquasecurity/trivy-java-db/pull/23/commits/5c3066767e8e5c8875fa4ddfe656362c116250ad
take a look, please.

Reply:
> database is downloaded once a week

Do you mean trivy-java-db? 

We update trivy-java-db every day:
https://github.com/aquasecurity/trivy-java-db/blob/15bfe3cde73658245ac47e7758ab2c6da5538bd6/.github/workflows/cron.yml#L5

Users download java-db every 3 days:
https://github.com/aquasecurity/trivy-java-db/blob/15bfe3cde73658245ac47e7758ab2c6da5538bd6/pkg/builder/builder.go#L20

But i understand your point.
I think it is possible. We can get wrong GroupID (if version is not added).
But i am not sure that our changes will cause many issues:
- after 1(3) days user will still see new package (perhaps we need to add info in Trivy docs about need to update trivy-java-db in this case).
- we check `pom.properties` and `MANIFEST.MF` files before accessing java-db.